docs/en/integrations/data-ingestion/etl-tools/apache-beam/README.md

```diff
@@ -0,0 +1,119 @@
+                        .addField(Schema.Field.of("age", Schema.FieldType.INT16).withNullable(true))
+                        .addField(Schema.Field.of("insertion_time", Schema.FieldType.DATETIME).withNullable(false))
+                        .build();
+
+
+        // Apply transforms to the pipeline.
+        PCollection<String> lines = p.apply("ReadLines", TextIO.read().from("src/main/resources/input.csv"));
+
+
+        PCollection<Row> rows = lines.apply("ConvertToRow", ParDo.of(new DoFn<String, Row>() {
+            @ProcessElement
+            public void processElement(@Element String line, OutputReceiver<Row> out) {
+            
+                String[] values = line.split(",");
+                Row row = Row.withSchema(SCHEMA)
+                        .addValues(values[0], Short.parseShort(values[1]), DateTime.now())
+                        .build();
+                out.output(row);
+            }
+        })).setRowSchema(SCHEMA);
+
+        rows.apply("Write to ClickHouse",
+                        ClickHouseIO.write("jdbc:clickhouse://localhost:8123/default?user=default&password=******", "test_table"));
+
+        // Run the pipeline.
+        p.run().waitUntilFinish();
+    }
+}
+
+```
+
+## Supported Data Types
+
+| ClickHouse                   | Apache Beam              |
+|------------------------------|--------------------------|
+| TableSchema.TypeName.FLOAT32 | Schema.TypeName#FLOAT    |
+| TableSchema.TypeName.FLOAT64 | Schema.TypeName#DOUBLE   |
+| TableSchema.TypeName.INT8    | Schema.TypeName#BYTE     |
+| TableSchema.TypeName.INT16   | Schema.TypeName#INT16    |
+| TableSchema.TypeName.INT32   | Schema.TypeName#INT32    |
+| TableSchema.TypeName.INT64   | Schema.TypeName#INT64    |
+| TableSchema.TypeName.STRING  | Schema.TypeName#STRING   |
+| TableSchema.TypeName.UINT8   | Schema.TypeName#INT16    |
+| TableSchema.TypeName.UINT16  | Schema.TypeName#INT32    |
+| TableSchema.TypeName.UINT32  | Schema.TypeName#INT64    |
+| TableSchema.TypeName.UINT64  | Schema.TypeName#INT64    |
+| TableSchema.TypeName.DATE    | Schema.TypeName#DATETIME |
+| TableSchema.TypeName.DATETIME| Schema.TypeName#DATETIME |
+| TableSchema.TypeName.ARRAY   | Schema.TypeName#ARRAY    |
+
+
+## Limitations
+
+Please consider the following limitations when using the connector:
+* The current supported ClickHouse JDBC version is `0.3.2-patch10`
+* As of today, only Sink operation is supported (the connector doesn't support Source operation)
+* ClickHouse performs deduplication when inserting into a `ReplicatedMergeTree` or a `Distributed` table built on top of a `ReplicatedMergeTree`. Without replication, inserting into a regular MergeTree can result in duplicates if an insert fails and then successfully retries. However, each block is inserted atomically, and the block size can be configured using `ClickHouseIO.Write.withMaxInsertBlockSize(long)`. Deduplication is achieved by using checksums of the inserted blocks.
```

Comment:
We can also provide a link https://clickhouse.com/docs/en/guides/developer/deduplication and https://clickhouse.com/docs/en/operations/settings/settings#insert-deduplicate for dedup reference